src/index.js

```diff
@@ -0,0 +1,219 @@
+      },
+    },
+  );
+}
+
+async function getCommentAPI(itemID) {
+  const response = await fetch(
+    `https://us-central1-involvement-api.cloudfunctions.net/capstoneApi/apps/gPGYyR5ezimXgm2rDsPh/comments?item_id=${itemID}`,
+  );
+  const jsonResponse = await response.json();
+  return jsonResponse;
+}
+
+function addCommentHTML(comment) {
+  const listItem = document.createElement('li');
+  listItem.classList.add('comment-list-item');
+  const commentName = document.createElement('p');
+  const commentText = document.createElement('p');
+  const commentDate = document.createElement('p');
+  commentName.innerText = `${comment.username}:`;
+  commentText.innerText = `${comment.comment}`;
+  commentDate.innerText = `${comment.creation_date}`;
+  listItem.append(commentName, commentText, commentDate);
+  document.body.querySelector('.comments-list').append(listItem);
+}
+
+async function commentSection(id) {
+  document.body.querySelector('.comments-list').innerHTML = '';
+  const comments = await getCommentAPI(id);
+  if (Array.isArray(comments)) {
+    comments.forEach((comment) => {
+      addCommentHTML(comment);
+    });
+    updateCounter(comments);
+    return;
+  }
+  document.body.querySelector('.comments-heading').innerText = 'Comments';
+}
+
+async function addComment(id) {
+  const name = document.body.querySelector('.form-name').value;
+  const comment = document.querySelector('.form-comment').value;
+  if (name === '' || comment === '') {
+    return;
+  }
+  await postCommentAPI(id, name, comment);
+  const commentsArray = await getCommentAPI(id);
+  const newComment = commentsArray[commentsArray.length - 1];
+  addCommentHTML(newComment);
+  updateCounter(commentsArray);
+  document.body.querySelector('.form-name').value = '';
+  document.querySelector('.form-comment').value = '';
+}
+
+async function recipeSection(id) {
+  const mealDetails = await getRecipeInfo(id);
+  document.body
+    .querySelector('.recipe-details')
+    .classList.remove('recipe-details-hide');
+  document.body.querySelector('.recipe-modal-heading').innerText = mealDetails.strMeal;
+  document.body.querySelector('.recipe-modal-image').src = mealDetails.strMealThumb;
+  document.body.querySelector('.recipe-modal-instructions').innerText = mealDetails.strInstructions;
+  document.body.querySelector('.recipe-modal-ingredients').innerHTML = '';
+  for (let i = 1; i < 20; i += 1) {
+    const ingredients = document.createElement('li');
+    const ingredient = `strIngredient${i}`;
+    const measurement = `strMeasure${i}`;
+    if (mealDetails[ingredient] === '') {
+      break;
+    }
+    ingredients.innerText = `${mealDetails[measurement]} ${mealDetails[ingredient]}`;
+    document.body
+      .querySelector('.recipe-modal-ingredients')
+      .append(ingredients);
+  }
+  document.body.querySelector('.add-comment-button').onclick = () => {
+    addComment(id);
+  };
+  commentSection(id);
+}
+
+async function generateHTML(category) {
```

Comment:
A few methods can be extracted out of this, and the method name could be `renderCategoryHTML` perhaps.